app.py

```diff
@@ -383,7 +385,8 @@ def modificar_usuario():
         cursor.execute("SELECT `usuario` FROM `my_resto`.`usuarios` ;")
         usuarios_backend = cursor.fetchall()
         usuarios_registrados = set(i[0] for i in usuarios_backend)
-        if not (nuevo_nombre in usuarios_registrados and nuevo_nombre == session['username']):
+        if not (nuevo_nombre in usuarios_registrados
```

Comment:
@gianfranco-s  Acá hay un error grave. 
Para ese condicional, hay 3 casos posibles(no 4 como uno esperaría en un condicional de 2 variables)
1) V - V  
2) V - F
no es posible el caso F-V ya que session['username'] es parte de usuarios_registrados
3) F - F

En el caso 1), lo que esta dentro del if, debería ejecutarse, ya que el nombre de usuario no fue modificado, pero puede haberse modificado el password, y no se ejecuta.
en el caso 2)No se deberia ejecutar el bloque, ya que en nuevo nombre ya esta en uso por OTRO usuario. y sin embargo se ejecuta.
Solo funciona de manera correcta en el caso 3.
La forma correcta de hacerlo, seria:
`if nuevo_nombre not in usuarios_registrados or nuevo_nombre == session['username']`